packages/typescript.rb

```diff
@@ -0,0 +1,43 @@
+# Adapted from Arch Linux typescript PKGBUILD at:
+# https://github.com/archlinux/svntogit-community/raw/packages/typescript/trunk/PKGBUILD
+
+require 'package'
+
+class Typescript < Package
+  description 'JavaScript with syntax for types'
+  homepage 'http://www.typescriptlang.org'
+  version '5.0.4'
+  license 'Apache'
+  compatibility 'all'
+  source_url 'https://github.com/microsoft/typescript.git'
+  git_hashtag "v#{version}"
+  binary_compression 'tar.zst'
+
+  binary_sha256({
+    aarch64: 'db9b20989f23947e00eec5f8bfd6181d5c41880227512e1033173b1ed6ced1af',
+     armv7l: 'db9b20989f23947e00eec5f8bfd6181d5c41880227512e1033173b1ed6ced1af',
+       i686: 'ceb26159248f454776599e094ebed584f3c6bb74963df861973937e3418f1078',
+     x86_64: '9ad883af55bc37a4f65f4719333736b15d0cbe953683a3a07806420e431affdf'
+  })
+
+  depends_on 'nodebrew' # L
+  depends_on 'rsync' => :build
+
+  def self.prebuild
+    system 'crew postinstall nodebrew'
+  end
+
+  def self.build
+    system 'npm ci'
+    system 'npx hereby LKG'
```

Comment:
Does this require a specific version of node/npm/npx?  I know any node version over 16 won't work with the lower glibc versions.

Reply:
The default versions run fine on lower glibc versions.

Reply:
What do you consider the "default" node version?  The latest is 25 which is not compatible with glibc 2.27 or lower.  Nodebrew will allow you to install whichever version you want.

Reply:
This PR has `nodebrew` install the last functioning node version for glibc 2.23 for i686 and 2.27.

But you can install whatever version you want afterwards! This just lets node run such that typescript binaries, which use node, work.